discovery-provider/src/utils/multi_provider.py

```diff
@@ -0,0 +1,21 @@
+import random
+from web3.providers import HTTPProvider, BaseProvider
+
+
+class MultiProvider(BaseProvider):
```

Comment:
document this class please :) similar to how you did the libs multiprovider
lets also be explicit about why we only need to define a subset of functions

Reply:
Added a docstring, with a link to the docs we reference.